src/include/ipc/pm.h

```diff
@@ -54,4 +54,20 @@ struct sof_ipc_pm_core_config {
 	uint32_t enable_mask;
 } __attribute__((packed));
 
+struct sof_ipc_pm_gate {
+	struct sof_ipc_cmd_hdr hdr;
+	uint32_t flags;
+} __attribute__((packed));
```

Comment:
Hi @mmaka1 can you help do a little bit change here:

The merged version in kernel side is like this:
```
struct sof_ipc_pm_gate {
	struct sof_ipc_cmd_hdr hdr;
	uint32_t flags;	/* platform specific */

+	/* reserved for future use */
+	uint32_t reserved[5];
} __packed;
```

Reply:
@keyonjie OK, I may added them if they are already merged into the kernel, however we should stop adding reserved fields. As long as you *append* new things and do not change syntax nor semantics of the existing ones and `struct sof_ipc_cmd_hdr::size` is set correctly, nothing prevents the handler to deserialize the data correctly from the communication channel.